debezium-testing/debezium-testing-system/src/test/resources/database-resources/mongodb/sharded/create-dbz-user.js

```diff
@@ -0,0 +1,30 @@
+// CREATE DBZ USER
+db = db.getSiblingDB('admin');
+db.runCommand({
+    createRole: "listDatabases",
+    privileges: [
+        { resource: { cluster : true }, actions: ["listDatabases"]}
+    ],
+    roles: []
+});
+
+db.runCommand({
+    createRole: "readChangeStream",
+    privileges: [
+        { resource: { db: "", collection: ""}, actions: [ "find", "changeStream" ] }
+    ],
+    roles: []
+});
+
+db.createUser({
+    user: 'debezium',
+    pwd: 'dbz',
+    roles: [
+        { role: "readWrite", db: "inventory" },
+        { role: "read", db: "local" },
+        { role: "listDatabases", db: "admin" },
+        { role: "readChangeStream", db: "admin" },
+        { role: "read", db: "config" },
+        { role: "read", db: "admin" }
```

Comment:
Note: Up until recent nightlies Debezium literary needed the ability to open change stream cluster wide. At that point you can just give it readAnyDatabase. However recently a functionality was added which allows running the connector with database restricted privileges.